src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-pg-scrub-modal/osd-pg-scrub-modal.component.ts

```diff
@@ -0,0 +1,73 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+
+import { I18n } from '@ngx-translate/i18n-polyfill';
+import { BsModalRef } from 'ngx-bootstrap/modal';
+import { forkJoin as observableForkJoin } from 'rxjs';
+
+import { ConfigOptionComponent } from '../../../../shared/components/config-option/config-option.component';
+import { ActionLabelsI18n } from '../../../../shared/constants/app.constants';
+import { NotificationType } from '../../../../shared/enum/notification-type.enum';
+import { CdFormGroup } from '../../../../shared/forms/cd-form-group';
+import { Permissions } from '../../../../shared/models/permissions';
+import { AuthStorageService } from '../../../../shared/services/auth-storage.service';
+import { NotificationService } from '../../../../shared/services/notification.service';
+import { OsdPgScrubModalOptions } from './osd-pg-scrub-modal.options';
+
+@Component({
+  selector: 'cd-osd-pg-scrub-modal',
+  templateUrl: './osd-pg-scrub-modal.component.html',
+  styleUrls: ['./osd-pg-scrub-modal.component.scss']
+})
+export class OsdPgScrubModalComponent implements OnInit {
+  osdPgScrubForm: CdFormGroup;
+  action: string;
+  resource: string;
+  permissions: Permissions;
+
+  @ViewChild('basicOptionsValues')
+  basicOptionsValues: ConfigOptionComponent;
+  basicOptions: Array<string> = OsdPgScrubModalOptions.basicOptions;
+
+  @ViewChild('advancedOptionsValues')
+  advancedOptionsValues: ConfigOptionComponent;
+  advancedOptions: Array<string> = OsdPgScrubModalOptions.advancedOptions;
+
+  advancedEnabled = false;
+
+  constructor(
+    public bsModalRef: BsModalRef,
+    private authStorageService: AuthStorageService,
+    private notificationService: NotificationService,
+    private i18n: I18n,
+    public actionLabels: ActionLabelsI18n
+  ) {
+    this.osdPgScrubForm = new CdFormGroup({});
+    this.resource = this.i18n('PG scrub options');
+    this.permissions = this.authStorageService.getPermissions();
+  }
+
+  ngOnInit() {
+    this.action = this.actionLabels.EDIT;
```

Comment:
Cannot say where better to put things: within `constructor()` or `ngOnInit()` hook, but given most was in constructor, you may move this assignment to constructor too (I haven't found any good reason on putting things in `constructor` vs. `ngOnInit` beyond appealing to tradition).

Reply:
It seems i18n is failing. You'll need to refresh messages.xlf after the actionLabel changes ;-)